### PR TITLE
RNMT-5718 - iOS 16 ::: Pull FLEX code from upstream

### PR DESCRIPTION
**Pull request by usernuno**
## Description

This PR pulls code from upstream and uses the latest version (`5.22.10`).

Changes that were kept:
- Remove extra options from toolbar and other menus
- Remove simulator keyboard shortcuts

Changes that were tweaked (due to changes between FLEX 3 and 5):
- Reduce toolbar size
- Remove string references to FLEX
- Toolbar button navigates directly to the Network Requests Panel (the only feature we allow)

Changes that were added (due to changes between FLEX 3 and 5):
- Change bottom bar buttons
- Remove host denylist related settings
- Rename OSCache to NLCache due to a naming conflict

A commit-by-commit review is advised.

Please also review the screenshots to find something that should be hidden, masked, or renamed.

P.S.: The `outsystems` branch was moved to be in line with the `5.22.10` tag. Please delete your local copy of this branch if you have it.

## Context

Fixes https://outsystemsrd.atlassian.net/browse/RNMT-5718

Reasoning for the update
- Reduce tech debt (we're 2 majors behind and using a version from 2019)
- There is a bug in FLEX. When targeting and running on iOS 16, opening the Network Requests Panel crashes the app.

This is fixed in version `5.22.10` of FLEX.

## Type of changes

- [x] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [x] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Tests

Manual tests were performed to check if all features works and if more UI changes are required.

## Screenshots (if appropriate)

Screenshot 2022-10-27 at 16 11 05
Screenshot 2022-10-27 at 16 11 14
Screenshot 2022-10-27 at 16 11 29
Screenshot 2022-10-27 at 16 11 50
Screenshot 2022-10-27 at 16 12 19
Screenshot 2022-10-27 at 16 12 25
Screenshot 2022-10-27 at 16 12 31
Screenshot 2022-10-27 at 16 13 08
Screenshot 2022-10-27 at 16 14 38
Screenshot 2022-10-27 at 16 15 01

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
    - [ ] Documentation has been updated accordingly

### CODE REVIEW

File: Classes/Core/Controllers/FLEXTableViewController.m [hunk 1]
```diff
@@ -324,16 +324,10 @@ - (void)setupToolbarItems {
...
-        UIBarButtonItem.flex_flexibleSpace,
-        self.bookmarksToolbarItem,
-        UIBarButtonItem.flex_flexibleSpace,
-        self.openTabsToolbarItem,
     ];
     
     for (UIBarButtonItem *item in self.toolbarItems) {
-        [item _setWidth:60];
-        // This does not work for anything but fixed spaces for some reason
-        // item.width = 60;
```
Comment (cmfsotelo) [inline Classes/Core/Controllers/FLEXTableViewController.m:336]: I love this comment on the source 😅

**Review by cmfsotelo — APPROVED** [review]
LGTM

Out of curiosity, what is the rationale for the NLCache name?

File: Classes/Core/Controllers/FLEXTableViewController.m (hunk 1 above)
Comment (usernuno) [reply Classes/Core/Controllers/FLEXTableViewController.m:336]: > Out of curiosity, what is the rationale for the NLCache name?

Nick Lockwood, the author of the OSCache.

**Comment by usernuno** [comment]
> Out of curiosity, what is the rationale for the NLCache name?

Nick Lockwood, the author of the OSCache.

**Review by tiagoappereira — APPROVED** [review]
LGTM

**Pull request merged**